src/ASM/LR/ASMu2Dnurbs.C

```diff
@@ -0,0 +1,339 @@
+// $Id$
+//==============================================================================
+//!
+//! \file ASMu2Dnurbs.C
+//!
+//! \date Nov 28 2018
+//!
+//! \author Arne Morten Kvarving / SINTEF
+//!
+//! \brief Driver for assembly of unstructured 2D NURBS FE models.
+//!
+//==============================================================================
+
+#include "GoTools/geometry/SplineSurface.h"
+
+#include "LRSpline/LRSplineSurface.h"
+#include "LRSpline/Element.h"
+#include "LRSpline/Basisfunction.h"
+
+#include "ASMu2Dnurbs.h"
+#include "FiniteElement.h"
+#include "CoordinateMapping.h"
+
+
+ASMu2Dnurbs::ASMu2Dnurbs (unsigned char n_s, unsigned char n_f)
+  : ASMu2D(n_s, n_f)
+{
+  noNurbs = false;
+}
+
+
+ASMu2Dnurbs::ASMu2Dnurbs (const ASMu2Dnurbs& patch, unsigned char n_f)
+  : ASMu2D(patch, n_f)
+{
+  noNurbs = patch.noNurbs;
+}
+
+
+bool ASMu2Dnurbs::read (std::istream& is)
+{
+  bool ok = this->ASMu2D::read(is);
+  if (ok && noNurbs)
+    std::cout <<"  ** LR-nurbs requested but input is a spline."<< std::endl;
+
+  return ok;
+}
+
+
+bool ASMu2Dnurbs::evaluateBasis (int iel, FiniteElement& fe,
+                                 int derivs) const
+{
+  if (noNurbs)
+    return this->ASMu2D::evaluateBasis(iel,fe,derivs);
+
+  const LR::Element* el = lrspline->getElement(iel);
+  if (!el) return false;
+
+  fe.xi  = 2.0*(fe.u - el->umin()) / (el->umax() - el->umin()) - 1.0;
+  fe.eta = 2.0*(fe.v - el->vmin()) / (el->vmax() - el->vmin()) - 1.0;
+  RealArray Nu = bezier_u.computeBasisValues(fe.xi, derivs);
+  RealArray Nv = bezier_v.computeBasisValues(fe.eta,derivs);
```

Comment:
In the corresponding ASMu2D-method you put a `#pragma omp critical` on these two statements. But I guess the same would apply here also? Or are they no longer needed. Should be consequent at least.